kai_mcp_solution_server/src/kai_mcp_solution_server/ast_diff/java.py

```
@@ -0,0 +1,411 @@
+            }
+        if self.classes != o.classes:
+            diff["classes"] = self.classes.diff(o.classes)
+        else:
+            diff["classes"] = list(self.classes.iter_values())
+        return diff
+
+
+def _extract_java_info(root: ts.Tree) -> DiffableSummary:
+    cursor = root.walk()
+
+    def traverse(node: ts.Node) -> DiffableSummary:
```

Pylint (3.3.7)

[refactor] Too many local variables (26/15)

(R0914)